### PR TITLE
Improve instructions for db-file fixtures

### PR DESCRIPTION
**Pull request by fblanchetNaN**
Improve `pytest.skip` messages when db-file are missing to point out at submodule qcodes_db_fixtures.

Explicitly recommend cloning with submodules, include the `git clone` command and options in the documentation.

Fix related error 404 in CONTRIBUTING.rst.

See QCoDeS/qcodes_generate_test_db#8

### CODE REVIEW

**Comment by jenshnielsen** [comment]
@fblanchetNaN Sorry it seemed like our prs have crossed. I am happy to merge yours, but I think it would be nice to do the refactor from mine to avoid duplication of the skip message

**Comment by fblanchetNaN** [comment]
> @fblanchetNaN Sorry it seemed like our prs have crossed. I am happy to merge yours, but I think it would be nice to do the refactor from mine to avoid duplication of the skip message

I rebased on your commit, adjusted my code to yours, and force-pushed my modifications.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -76,7 +76,7 @@ Setup
 ~~~~~
 
 -  Clone and register the package for development as described in the
-   `README <README.md#installation>`__
+   `here <http://qcodes.github.io/Qcodes/start/index.html#installation>`__
```
Comment (jenshnielsen) [inline CONTRIBUTING.rst:79]: here does not seem to match the line above.

**Review by jenshnielsen — APPROVED** [review]

File: CONTRIBUTING.rst (hunk 1 above)
Comment (fblanchetNaN) [reply CONTRIBUTING.rst]: I removed "in the" and keep "here" for the link to be coherent with the style of README.rst.

**Review by jenshnielsen — APPROVED** [review]
Bors merge

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**